### PR TITLE
Add support to migrate a directory.

### PR DESCRIPTION
**Pull request by Sridhar-MS**
- Given a directory migrate all the project.jsons in it recursively.
- Remove '-p/--project' option. Instead directly pass a directory or project.json as argument.
- Add tests.

Fixes 1 and 3 in #4167 

cc @brthor @livarcocc @eerhardt

### CODE REVIEW

File: src/dotnet/commands/dotnet-migrate/MigrateCommand.cs [hunk 1]
```diff
@@ -32,24 +33,48 @@ public MigrateCommand(string templateFile, string projectJson, string sdkVersion
...
-            var outputDirectory = projectDirectory;
-
             var sdkVersion = _sdkVersion ?? new ProjectJsonParser(_temporaryDotnetNewProject.ProjectJson).SdkPackageVersion;
             EnsureNotNull(sdkVersion, "Null Sdk Version");
 
-            var migrationSettings = new MigrationSettings(projectDirectory, outputDirectory, sdkVersion, msBuildTemplate, _xprojFilePath);
-            new ProjectMigrator().Migrate(migrationSettings, _skipProjectReferences);
+            foreach (var project in projectsToMigrate)
+            {
+                Console.WriteLine($"START migrating project {project}..");
```
Comment (brthor) [inline src/dotnet/commands/dotnet-migrate/MigrateCommand.cs]: Make this a trace call?

File: test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs [hunk 2]
```diff
@@ -150,13 +145,36 @@ public void It_migrates_root_project_and_skips_references(string projectName)
 
             FixUpProjectJsons(projectDirectory);
 
-            MigrateCommand.Run(new [] { "-p", Path.Combine(projectDirectory, projectName), "--skip-project-references" }).Should().Be(0);
+            MigrateCommand.Run(new [] { Path.Combine(projectDirectory, projectName), "--skip-project-references" }).Should().Be(0);
```
Comment (brthor) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs]: Can you add a test combining recursion and not skipping project refs?

File: src/dotnet/commands/dotnet-migrate/MigrateCommand.cs (hunk 1 above)
Comment (brthor) [inline src/dotnet/commands/dotnet-migrate/MigrateCommand.cs]: At least reduce to one message, "Migrated project {project}" rather than start/end

Comment (Sridhar-MS) [inline src/dotnet/commands/dotnet-migrate/MigrateCommand.cs]: Fixed

File: test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs (hunk 2 above)
Comment (Sridhar-MS) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs]: Fixed

File: src/dotnet/commands/dotnet-migrate/MigrateCommand.cs [hunk 3]
```diff
@@ -32,24 +33,47 @@ public MigrateCommand(string templateFile, string projectJson, string sdkVersion
...
+                var projects = Directory.EnumerateFiles(projectArg, Project.FileName, SearchOption.AllDirectories);
+
+                foreach(var project in projects)
+                {
+                    yield return GetProjectJsonPath(project);
+                }
+            }
+            else
+            {
+                throw new Exception($"Invalid project argument - {projectArg}");
```
Comment (eerhardt) [inline src/dotnet/commands/dotnet-migrate/MigrateCommand.cs]: Can we make this error a little more informative?  Something like "'{projectArg}' is not a project.json file and a directory named '{projectArg}' doesn't exist".

**Comment by eerhardt** [comment]
:shipit:

File: src/dotnet/commands/dotnet-migrate/MigrateCommand.cs (hunk 3 above)
Comment (Sridhar-MS) [inline src/dotnet/commands/dotnet-migrate/MigrateCommand.cs]: Fixed

**Pull request merged**